src/AuthProvider.jsx

```diff
@@ -0,0 +1,36 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+const AuthContext = createContext();
+
+export const useAuth = () => {
+    return useContext(AuthContext);
+};
+
+export const AuthProvider = ({ children}) => {
+    const [isLoggedIn, setIsLoggedIn] = useState(false);
+    const [userData, setUserData] = useState({ firstName: "", lastName: "", role: "" })
+    const isToken = sessionStorage.getItem('isToken');
```

Comment:
The actual token should be kept available in sessions for any further validation requirement.

Reply:
Did you mean that the token value not the boolean should be stored?

Reply:
yes